main.go

```diff
@@ -0,0 +1,242 @@
+
+	"github.com/operator-framework/operator-lib/leader"
+
+	routev1 "github.com/openshift/api/route/v1"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/openshift/operator-custom-metrics/pkg/metrics"
+	aaov1alpha1 "github.com/ravitri/aws-account-operator/api/v1alpha1"
+
+	certmanv1alpha1 "github.com/openshift/certman-operator/api/v1alpha1"
+	operatorconfig "github.com/openshift/certman-operator/config"
+	"github.com/openshift/certman-operator/controllers/certificaterequest"
+	"github.com/openshift/certman-operator/controllers/clusterdeployment"
+	cClient "github.com/openshift/certman-operator/pkg/clients"
+	"github.com/openshift/certman-operator/pkg/k8sutil"
+	"github.com/openshift/certman-operator/pkg/localmetrics"
+	"github.com/openshift/certman-operator/pkg/version"
+	//+kubebuilder:scaffold:imports
+)
+
+var (
+	hours       int = 4
+	metricsPath     = "/metrics"
+	metricsPort     = "8080"
+	scheme          = apiruntime.NewScheme()
+	setupLog        = ctrl.Log.WithName("setup")
+)
+
+var log = logf.Log.WithName("cmd")
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(certmanv1alpha1.AddToScheme(scheme))
+	utilruntime.Must(routev1.Install(scheme))
+	utilruntime.Must(hivev1.AddToScheme(scheme))
+	utilruntime.Must(aaov1alpha1.AddToScheme(scheme))
+	//+kubebuilder:scaffold:scheme
+}
+
+func printVersion() {
+	log.Info(fmt.Sprintf("Go Version: %s", runtime.Version()))
+	log.Info(fmt.Sprintf("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH))
+	log.Info(fmt.Sprintf("Version of operator-sdk: %v", version.SDKVersion))
+}
+
+func main() {
+	var metricsAddr string
+	var enableLeaderElection bool
+	var probeAddr string
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":"+metricsPort, "The address the metric endpoint binds to.")
+	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
+	opts := zap.Options{
+		Development: true,
+	}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+
+	// Add a custom logger to log in RFC3339 format instead of UTC
+	configLog := uzap.NewProductionEncoderConfig()
+	configLog.EncodeTime = func(ts time.Time, encoder zapcore.PrimitiveArrayEncoder) {
+		encoder.AppendString(ts.UTC().Format(time.RFC3339Nano))
+	}
+	logfmtEncoder := zaplogfmt.NewEncoder(configLog)
+	logger := zap.New(zap.UseDevMode(true), zap.WriteTo(os.Stdout), zap.Encoder(logfmtEncoder))
+	logf.SetLogger(logger)
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+
+	printVersion()
+
+	namespace, err := k8sutil.GetWatchNamespace()
+	if err != nil {
+		log.Error(err, "Failed to get watch namespace")
+		os.Exit(1)
+	}
+
+	if err := routev1.AddToScheme(clientgoscheme.Scheme); err != nil {
+		log.Error(err, "unable to add routev1 scheme")
+		os.Exit(1)
+	}
```

Comment:
A minor nit, but if `operator-custom-metrics` dependency is bumped up to at least `v0.5.1-0.20220802235640-dc76a1f15ee8` or greater then it should prevent the need for this. (I'm assuming this was added to allow the operator to create the metrics route successfully)

Reply:
Thanks a lot Matt! Yup, have updated the dependency and tested it as well. Seemed to work fine and certman-operator didn't complain of the route API being unknown.